### PR TITLE
edit modal for user's roles

### PR DESCRIPTION
**Pull request by gerard-bisama**

### CODE REVIEW

File: server/bin/www [hunk 1]
```diff
@@ -7,18 +7,25 @@
...
+const appConfig = require('../config/config');
+const winston = require('winston');
+
 
 /**
  * Get port from environment and store in Express.
  */
 
-var port = normalizePort(process.env.PORT || '3000');
+//var port = normalizePort(process.env.PORT || '3000');
```
Comment (lokisapocalypse) [inline server/bin/www]: Commented code like this needs to be removed.

File: server/config/config.json.example [hunk 2]
```diff
@@ -1,4 +1,5 @@
 {
+  "port":3000,
```
Comment (lokisapocalypse) [inline server/config/config.json.example]: The port should be inside development, staging, etc.

File: server/routes/user.js [hunk 3]
```diff
@@ -27,7 +27,8 @@ router.post("/add", function (req, res, next) {
     id: "user",
     extension: [
       {
-        url: config.structureDefinition + "/StructureDefinition/iHRISUserDetails",
+        //url: config.structureDefinition + "/StructureDefinition/iHRISUserDetails",
```
Comment (lokisapocalypse) [inline server/routes/user.js]: Remove the commented code

File: server/routes/user.js [hunk 4]
```diff
@@ -83,7 +91,29 @@ router.get("/list", function (req, res, next) {
...
+    if (err) {
+      res.status(400).json(err);
+    } else {
+      response=definition;
+      rolesDefinition=null;
+      response.differential.element.forEach(extension=>{
+        if(extension.id=="Extension.extension:roles.value[x]")
+        {
+          rolesDefinition=extension;
+          //break;
```
Comment (lokisapocalypse) [inline server/routes/user.js]: Remove commented code.

File: server/routes/user.js [hunk 5]
```diff
@@ -83,7 +91,29 @@ router.get("/list", function (req, res, next) {
...
+        }
+      });
+      //res.status(201).json(definition);
```
Comment (lokisapocalypse) [inline server/routes/user.js]: Remove commented code.

File: server/routes/user.js [hunk 6]
```diff
@@ -254,4 +284,26 @@ router.post("/update", function (req, res, next) {
   });
 });
 
+/**
+ * update user role
+ */
+router.post("/updaterole", function (req, res, next) {
```
Comment (lokisapocalypse) [inline server/routes/user.js]: The convention we've been using for route names is words separated by hyphens. So "update-role" instead of updaterole.

File: ui/vue.config.js [hunk 7]
```diff
@@ -1,5 +1,7 @@
+const appConfig = require('./src/config/config');
 module.exports = {
   devServer: {
-    proxy: "http://localhost:3000"
+    //proxy: "http://localhost:3001"
```
Comment (lokisapocalypse) [inline ui/vue.config.js]: Remove this commented code as well.

File: ui/src/views/Admin/AddUser.vue [hunk 8]
```diff
@@ -24,43 +25,87 @@ import axios from "axios";
 
 import Alert from "@/components/Layout/Alert.vue";
 import DynamicForm from "@/components/Form/DynamicForm.vue";
-
+//var tempFlields=[];
```
Comment (lokisapocalypse) [inline ui/src/views/Admin/AddUser.vue]: Remove this commented line.

File: ui/src/views/Admin/AddUser.vue [hunk 9]
```diff
@@ -24,43 +25,87 @@ import axios from "axios";
...
 export default {
   components: {
     Alert,
     DynamicForm
   },
-  created() {
+  created () {
+    NProgress.start();
     this.config = require("@/config/config.json");
+    var tempFlields=[{
```
Comment (lokisapocalypse) [inline ui/src/views/Admin/AddUser.vue]: Why is this here instead of inside data()?

File: server/config/config.json.example (hunk 2 above)
Comment (gerard-bisama) [reply server/config/config.json.example]: Done

File: server/routes/user.js (hunk 3 above)
Comment (gerard-bisama) [reply server/routes/user.js]: Done
